### ISSUE TITLE
ts.getPreProcessedFileInfo doesn't detect imported files from import(`template`) and require(`tempalte`)

### ISSUE THREAD

**Issue by IllusionMH** [microsoft/TypeScript#33680]
**TypeScript Version:**  3.6.3, 3.7.0-dev.20190928

**Search Terms:**

**Code**

```ts
const ts = require('typescript');

const lines = [
    'const rs = require("rs.ts");',
    'const rt = require(`rt.ts`);',
    'const is = import("is.ts");',
    'const it = import(`it.ts`);',
].join('\n')

console.log(ts.preProcessFile(lines, true, true));
```

**Expected behavior:**

```js
{
    referencedFiles: [],
    typeReferenceDirectives: [],
    libReferenceDirectives: [],
    importedFiles: [
        { fileName: 'rs.ts', pos: 19, end: 24 },
        { fileName: 'rt.ts', pos: 48, end: 53 },
        { fileName: 'is.ts', pos: 76, end: 81 },
        { fileName: 'it.ts', pos: 104, end: 109 }
    ],
    isLibFile: false,
    ambientExternalModules: undefined
}
```

**Actual behavior:**

```js
{
    referencedFiles: [],
    typeReferenceDirectives: [],
    libReferenceDirectives: [],
    importedFiles: [
        { fileName: 'rs.ts', pos: 19, end: 24 },
        { fileName: 'is.ts', pos: 76, end: 81 }
    ],
    isLibFile: false,
    ambientExternalModules: undefined
}
```

Because `src/services/preProcess.ts` only checks for `StringLiteral`'s bun not for `NoSubstitutionTemplateLiteral`'s which were allowed there for a long time.

**Playground Link:** N/A

**Related Issues:** found during implementing #32064 , but this syntax and type checked worked long before this PR

**Comment by joscha** [comment]
fix in #36546
